mr_project/mr_app/views.py

```diff
@@ -1,9 +1,231 @@
+        print(user_name)
+
+
+def register_emotion(driver_arg, username, emotion):
+    epoch_timestamp = time()
+    with driver_arg.session() as session:
+        seconds = session.write_transaction(create_epoch_relationship_between_user_and_emotion, emotion, username, epoch_timestamp)
+        print(seconds)
+
+
+def spawn_all_emotions(driver_arg, emotions_list):
+    n_emotions = len(emotions_list)
+    for i in range(n_emotions):
+        spawn_emotion_node(driver_arg, emotions_list[i])
+
+
+def add_users(driver_arg, users_list):
+    n_users = len(users_list)
+    for i in range(n_users):
+        spawn_user_node(driver_arg, users_list[i])
+
+
+@csrf_exempt
+def get_user_emotion(request):
+    if request.method == 'GET':
+        query_arguments = request.GET
+        user = ''
+        emotion = ''
+
+        try:
+            user = query_arguments['user']
+            emotion = query_arguments['emotion']
+        except KeyError:
+            return HttpResponseBadRequest('Missing required key')
+
+        if check_for_user(DRIVER, user) and check_for_emotion(DRIVER, emotion):
+            epoch_dict = fetch_epochs_of_user_emotion_pair(DRIVER, user, emotion)
+
+            return JsonResponse(epoch_dict)
+        return HttpResponseBadRequest('Slow down dawg! The user and/or emotion do not exist, ya need a spelling lesson?')
+
+    return HttpResponseNotFound()
+
+
+@csrf_exempt
+def get_all_user_emotions(request):
+    if request.method == 'GET':
+        query_arguments = request.GET
+        user = ''
+
+        try:
+            user = query_arguments['user']
+        except KeyError:
+            return HttpResponseBadRequest('Missing required key')
+
+        if check_for_user(DRIVER, user):
+            all_epochs_dict = fetch_all_emotions_and_epochs_of_user(DRIVER, user)
+
+            return JsonResponse(all_epochs_dict)
+        return HttpResponseBadRequest('Slow down dawg! The user does not exist, ya need a spelling lesson?')
+
+    return HttpResponseNotFound()
+
+
+@csrf_exempt
+def create_user_emotion(request):
+    if request.method == 'POST':
+        body = json.loads(request.body)
+        user = ''
+        emotion = ''
+
+        try:
+            user = body['user']
+            emotion = body['emotion']
+        except KeyError:
+            return HttpResponseBadRequest('Missing required key')
+
+        if check_for_user(DRIVER, user) and check_for_emotion(DRIVER, emotion):
+            register_emotion(DRIVER, user, emotion)
+        elif check_for_user(DRIVER, user):
+            spawn_emotion_node(DRIVER, emotion)
+            register_emotion(DRIVER, user, emotion)
+        elif check_for_emotion(DRIVER, emotion):
+            spawn_user_node(DRIVER, user)
+            register_emotion(DRIVER, user, emotion)
+        else:
+            spawn_user_node(DRIVER, user)
+            spawn_emotion_node(DRIVER, emotion)
+            register_emotion(DRIVER, user, emotion)
```

Comment:
This can be simplified
```python
if not check_for_user(DRIVER, user):
  spawn_user_node(DRIVER, user)

if not check_for_emotion(DRIVER, emotion):
  spawn_emotion_node(DRIVER, emotion)

register_emotion(DRIVER, user, emotion)
```

Reply:
check